python/ray/monitor.py

```diff
@@ -178,14 +181,8 @@ def autoscaler_resource_request_handler(self, _, data):
             data: a resource request as JSON, e.g. {"CPU": 1}
         """
 
-        if not self.autoscaler:
-            return
-
-        try:
-            self.autoscaler.request_resources(json.loads(data))
-        except Exception:
-            # We don't want this to kill the monitor.
-            traceback.print_exc()
+        resource_request = json.loads(data)
+        self.load_metrics.set_resource_requests(resource_request)
```

Comment:
Why not keep the try-except in case the assert breaks it?

Reply:
I originally added that try/catch because we weren't sure how stable the resource demand reporting was, and didn't want to crash the legacy autoscaler due to the experimental feature. 

None of those original conditions are true anymore, so the try/catch should be unnecessary.

Reply:
I am not sure I understand. It seems like it still does the same thing, since it only included a call to a function that sets the resource_demand_vector, which it is doing now, it seems to me like the only thing this try except was trying to avoid is a wrong type being assigned to the resource demand vector. My concern is that autoscaler will crash if the user accidentally sets the wrong type for request_resources.

Reply:
> user accidentally sets 

This isn't possible because the sdk only publishes json lists to the channel. 

All bets are off if the user messes with the channel themselves.

Reply:
In that case, can you get rid of the assert inside request_resources()?